src/libcore/time.rs

```diff
@@ -501,13 +502,77 @@ impl Mul<u32> for Duration {
     }
 }
 
+#[stable(feature = "duration_mul_div_extras", since = "1.29.0")]
+impl Mul<Duration> for u32 {
+    type Output = Duration;
+
+    fn mul(self, rhs: Duration) -> Duration {
+        rhs.checked_mul(self).expect("overflow when multiplying scalar by duration")
```

Comment:
Maybe just use `rhs * self` to avoid repeating things?

Reply:
They have different panic messages "multiplying scalar by duration" vs "multiplying duration by scalar"

Reply:
If having different panic messages is not important, then I'll change it and `f64` impl to `rhs * self`.